File: translate.js
const APPID = '20230811001777285';
const KEY = 'LZCKVoFJXyWeoMr6BgC0';

function generateSign(query, salt) {
    const str = APPID + query + salt + KEY;
    return MD5(str);
}

async function translate(text) {
    const salt = new Date().getTime();
    const sign = generateSign(text, salt);
    
    const url = new URL('https://fanyi-api.baidu.com/api/trans/vip/translate');
    const params = {
        q: text,
        from: 'auto',
        to: 'zh',
        appid: APPID,
        salt: salt,
        sign: sign
    };
    
    url.search = new URLSearchParams(params).toString();
    
    try {
        const response = await fetch(url);
        const data = await response.json();
        
        if (data.error_code) {
            throw new Error(`翻译错误: ${data.error_msg}`);
        }
        return data.trans_result[0].dst;
    } catch (error) {
        throw error;
    }
}

chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
    if (request.action === 'translate') {
        translate(request.text)
            .then(result => sendResponse({result}))
            .catch(error => sendResponse({error: error.message}));
        return true;
    }
}); 

File: translate.py
import hashlib
import requests
import random
import json

def translate_text(text, appid, key, from_lang='auto', to_lang='zh'):
    salt = str(random.randint(32768, 65536))
    sign = appid + text + salt + key
    sign = hashlib.md5(sign.encode()).hexdigest()
    
    url = 'https://fanyi-api.baidu.com/api/trans/vip/translate'
    params = {
        'q': text,
        'from': from_lang,
        'to': to_lang,
        'appid': appid,
        'salt': salt,
        'sign': sign
    }
    
    try:
        response = requests.get(url, params=params)
        result = response.json()
        
        if 'error_code' in result:
            return f"错误: {result['error_msg']}"
            
        return result['trans_result'][0]['dst']
        
    except Exception as e:
        return f"翻译出错: {str(e)}"

if __name__ == '__main__':
    APPID = '20230811001777285'
    KEY = 'LZCKVoFJXyWeoMr6BgC0'
    
    text = input('请输入要翻译的文本: ')
    result = translate_text(text, APPID, KEY)
    print(f'翻译结果: {result}') 